zeus/optimizer/batch_size/server/batch_size_state/repository.py

```diff
@@ -0,0 +1,244 @@
+                    batch_size=exps[0].batch_size,
+                    explorations=deepcopy(exps),
+                )
+
+                exps = [ExplorationStateModel.from_orm(exp)]
+        if len(exps) != 0:
+            explorations_per_bs[exps[0].batch_size] = ExplorationsPerBs(
+                job_id=job_id,
+                batch_size=exps[0].batch_size,
+                explorations=deepcopy(exps),
+            )
+
+        return ExplorationsPerJob(
+            job_id=job_id, explorations_per_bs=explorations_per_bs
+        )
+
+    async def get_arms(self, job_id: UUID) -> list[GaussianTsArmStateModel]:
+        """Retrieve Gaussian Thompson Sampling arms for a given job.
+
+        Args:
+            job_id (UUID): The ID of the job.
+
+        Returns:
+            List[GaussianTsArmStateModel]: List of Gaussian Thompson Sampling arms. These arms are all "good" arms (converged during pruning stage).
+            Refer to `GaussianTsArmStateModel`[zeus.optimizer.batch_size.server.batch_size_state.models.GaussianTsArmStateModel] for attributes.
+        """
+        stmt = select(GaussianTsArmState).where(GaussianTsArmState.job_id == job_id)
+        res = (await self.session.scalars(stmt)).all()
+        return [GaussianTsArmStateModel.from_orm(arm) for arm in res]
+
+    async def get_arm(self, bs: BatchSizeBase) -> GaussianTsArmStateModel | None:
+        """Retrieve Gaussian Thompson Sampling arm for a given job id and batch size.
+
+        Args:
+            bs (BatchSizeBase): The batch size object.
+
+        Returns:
+            Optional[GaussianTsArmStateModel]: Gaussian Thompson Sampling arm if found, else None.
+            Refer to `GaussianTsArmStateModel`[zeus.optimizer.batch_size.server.batch_size_state.models.GaussianTsArmStateModel] for attributes.
+        """
+        stmt = select(GaussianTsArmState).where(
+            and_(
+                GaussianTsArmState.job_id == bs.job_id,
+                GaussianTsArmState.batch_size == bs.batch_size,
+            )
+        )
+        arm = await self.session.scalar(stmt)
+        if arm is None:
+            return None
+        return GaussianTsArmStateModel.from_orm(arm)
+
+    def add_exploration(self, exploration: CreateExploration) -> None:
+        """Add an exploration to the data base.
+
+        Refer to `CreateExploration`[zeus.optimizer.batch_size.server.batch_size_state.models.CreateExploration] for attributes.
+
+        Args:
+            exploration (CreateExploration): The exploration to add.
+        """
+        self.session.add(exploration.to_orm())
+
+    async def update_exploration(self, updated_exp: UpdateExploration) -> None:
+        """Update an exploration in the database.
+
+        Args:
+            updated_exp (UpdateExploration): The updated exploration.
+            Refer to `UpdateExploration`[zeus.optimizer.batch_size.server.batch_size_state.models.UpdateExploration] for attributes.
+        """
+        stmt = (
```

Comment:
BTW checking all the SQL statements was tough so I... skipped many of them. Please make sure they're good.